src/Plot.elm

```diff
@@ -0,0 +1,107 @@
+module Plot where
+
+import Svg exposing (svg, Svg)
+import Private.Extras.SvgAttributes exposing (width, height)
+import Private.Dimensions exposing (Dimensions)
+import Private.Margins as Margins exposing (Margins)
+import Private.BoundingBox as BoundingBox exposing (BoundingBox)
+import Private.Scale.Utils as Scale
+import Private.Scale exposing (Scale)
+import Plot.Axis as Axis exposing (Axis)
+import Private.Bars as Bars exposing (Orient)
+import Private.Axis.View as AxisView
+import Private.Title as Title
+
+type alias Plot =
+  { dimensions: Dimensions
+  , margins: Margins
+  , svgs: List (BoundingBox -> List Svg)
+  , eventHandlers: List (BoundingBox -> Svg.Attribute)
+  , attrs: List Svg.Attribute
+  , title : Title.Model
+  }
+
+createPlot : Float -> Float -> Plot
+createPlot w h =
+  { dimensions = { width = w, height = h }
+  , margins = Margins.init
+  , svgs = []
+  , eventHandlers = []
+  , attrs = [width w, height h]
+  , title = Title.init
+  }
+
+addTitle : String -> List Svg.Attribute -> Plot -> Plot
+addTitle title attrs plot =
+  { plot | title = Title.create title attrs }
+
+attributes : List Svg.Attribute -> Plot -> Plot
+attributes attrs plot =
+  { plot | attrs = plot.attrs ++ attrs }
+
+margins : Margins -> Plot -> Plot
+margins m plot =
+  { plot | margins = m }
+
+addVerticalBars : List a -> (a -> b) -> (a -> c) -> Scale x b -> Scale y c -> List Svg.Attribute -> Plot -> Plot
```

Comment:
The arguments for the scales, Scale x b -> Scale y c, are also going to travel together frequently (especially in the function addSvgWithTwoScales) where it could also benefit from a concrete type to wrap them up.